Bottom: bea1d9b0d2d853009505bf65672cf13cc8f460d2
Top:    e1327a3843349e9f747aa65de7a3f12f0fc3ac9a
Author: Allen Short <ashort@mozilla.com>
Date:   2017-09-14 06:23:44 +0000

Switch to PyMySQL for MySQL 5.7 support


---
diff --git a/redash/query_runner/mysql.py b/redash/query_runner/mysql.py
index bfd6e719..9c674aea 100644
--- a/redash/query_runner/mysql.py
+++ b/redash/query_runner/mysql.py
@@ -90,7 +90,7 @@ class Mysql(BaseSQLQueryRunner):
     @classmethod
     def enabled(cls):
         try:
-            import MySQLdb
+            import pymysql
         except ImportError:
             return False
 
@@ -126,11 +126,11 @@ class Mysql(BaseSQLQueryRunner):
         return schema.values()
 
     def run_query(self, query, user):
-        import MySQLdb
+        import pymysql
 
         connection = None
         try:
-            connection = MySQLdb.connect(host=self.configuration.get('host', ''),
+            connection = pymysql.connect(host=self.configuration.get('host', ''),
                                          user=self.configuration.get('user', ''),
                                          passwd=self.configuration.get('passwd', ''),
                                          db=self.configuration['db'],
@@ -160,7 +160,7 @@ class Mysql(BaseSQLQueryRunner):
                 error = "No data was returned."
 
             cursor.close()
-        except MySQLdb.Error as e:
+        except pymysql.Error as e:
             json_data = None
             error = e.args[1]
         except KeyboardInterrupt:
diff --git a/requirements_all_ds.txt b/requirements_all_ds.txt
index f40320cc..bc1b64c1 100644
--- a/requirements_all_ds.txt
+++ b/requirements_all_ds.txt
@@ -2,7 +2,7 @@ google-api-python-client==1.5.1
 gspread==0.6.2
 impyla==0.10.0
 influxdb==2.7.1
-MySQL-python==1.2.5
+PyMySQL==0.7.11
 oauth2client==3.0.0
 pyhive==0.5.1
 pymongo[tls,srv]==3.6.1
